tlslite/tlsrecordlayer.py

```diff
@@ -722,6 +736,40 @@ def _getMsg(self, expectedType, secondaryType=None, constructorType=None):
                                 yield result
                             continue
 
+                    # If we received a heartbeat request and heartbeat
+                    # extension was negotiated
+                    if recordHeader.type == ContentType.heartbeat and \
+                            self.heartbeatSupported:
+                        try:
+                            heartbeat_message = Heartbeat().parse(p)
+                            # If we received heartbeat request, then we
+                            # response with response create from request
+                            if heartbeat_message.message_type == \
+                                    HeartbeatMessageType.heartbeat_request:
+                                if not self.heartbeatCanReceive:
+                                    for result in self._sendError(
+                                            AlertDescription.
+                                            unexpected_message,
+                                            "Received heartbeat_request to " +
+                                            "peer_not_allowed_to_send mode"):
+                                        yield result
+                                else:
+                                    heartbeat_response = heartbeat_message.\
+                                        createResponse()
+                                for result in self._sendMsg(
+                                        heartbeat_response):
+                                    yield result
+                            # If we received heartbeat response, then we
+                            # check, if its payload is same as payload of
+                            # request we sent
+                            elif heartbeat_message.message_type == \
+                                    HeartbeatMessageType.heartbeat_response:
+                                heartbeat_message.checkResponse(
```

Comment:
What do you mean? Directly calling e.g. `self.heartbeatReponceCallback(heartbeat_message.payload)`? I´m not sure about it and how callback can look like (what parameters it will have).